### PR TITLE
GI-162: QuizFlow korrigieren und optimieren

### PR DESCRIPTION
**Pull request by noadjamila**
## Description

In this PR, I revised the QuizFlow so that an error-free and correct quiz is guaranteed. 
It now queries the correct database and the match between the user's riasec scores and those of the degree programs is calculated using a cosine similarity. Data transfer between levels (study IDs) has been corrected and improved. 

## Jira Ticket

Link to the Jira ticket: [GI-162](https://techstudyfinder.atlassian.net/browse/GI-162)

## Author Checklist

- [x] Code follows project coding standards and conventions
- [x] Tests have been written or updated
- [x] All tests pass locally
- [x] No sensitive data or secrets are included
- [x] Documentation (README, comments) has been updated if necessary

## Reviewer Notes

Please verify the following during review:

- The implementation matches the described requirements and Jira ticket
- Code changes are logically correct and maintainable
- No unnecessary dependencies or side effects introduced
- Tests cover critical logic and edge cases
- Naming, structure, and readability meet team standards

### CODE REVIEW

File: client/src/components/quiz/Quiz_L1.tsx [hunk 1]
```diff
@@ -38,8 +40,8 @@ export default function QuizPage_L1({ onNextLevel }: QuizPageL1Props) {
           level: 1,
           answers: answersPayload,
         });
-
-        onNextLevel?.(res.ids);
+        console.log("Filtered IDs from Level 1:", res.ids);
```
Comment (psophis) [inline client/src/components/quiz/Quiz_L1.tsx]:
```suggestion
```

**Comment by noadjamila** [comment]
Erledigt :) @psophis

**Review by psophis — APPROVED** [review]
Danke für den Clean-Up und die Strukturierung! 💗
